spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/SkipperStreamService.java

```diff
@@ -0,0 +1,259 @@
+		this.appRegistryService = appRegistryService;
+		this.skipperStreamDeployer = skipperStreamDeployer;
+		this.appDeploymentRequestCreator = appDeploymentRequestCreator;
+	}
+
+	/**
+	 * Deploy a stream as defined by its stream name and optional deployment properties.
+	 *
+	 * @param streamName the stream name to deploy
+	 * @param deploymentProperties the deployment properties for the stream
+	 */
+	@Override
+	public void doDeployStream(String streamName, Map<String, String> deploymentProperties) {
+
+		StreamDefinition streamDefinition = createStreamDefinitionForDeploy(streamName);
+
+		// Extract skipper properties
+		Map<String, String> skipperDeploymentProperties = getSkipperProperties(deploymentProperties);
+		// Create map without any skipper properties
+		Map<String, String> deploymentPropertiesToUse = deploymentProperties.entrySet().stream()
+				.filter(mapEntry -> !mapEntry.getKey().startsWith(SKIPPER_KEY_PREFIX))
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+		List<AppDeploymentRequest> appDeploymentRequests = this.appDeploymentRequestCreator
+				.createRequests(streamDefinition, deploymentPropertiesToUse);
+
+		DeploymentPropertiesUtils.validateSkipperDeploymentProperties(deploymentPropertiesToUse);
+		Release release = this.skipperStreamDeployer.deployStream(new StreamDeploymentRequest(streamDefinition.getName(),
+				streamDefinition.getDslText(), appDeploymentRequests, skipperDeploymentProperties));
+
+		if (release != null) {
+			updateStreamDefinitionFromReleaseManifest(streamName, release.getManifest());
+		}
+		else {
+			logger.warn("Missing release after Stream Update!");
+		}
+	}
+
+	@Override
+	public String doCalculateStreamState(String name) {
+		return this.skipperStreamDeployer.calculateStreamState(name);
+	}
+
+	@Override
+	public void doUndeployStream(String streamName) {
+		this.skipperStreamDeployer.undeployStream(streamName);
+	}
+
+	@Override
+	public void doUpdateStream(String streamName, String releaseName, PackageIdentifier packageIdentifier, Map<String,
+			String> updateProperties) {
+		String yamlProperties = convertPropertiesToSkipperYaml(streamName, updateProperties);
+		Release release = this.skipperStreamDeployer.upgradeStream(releaseName, packageIdentifier, yamlProperties);
+
+		if (release != null) {
+			updateStreamDefinitionFromReleaseManifest(streamName, release.getManifest());
+		}
+		else {
+			logger.warn("Missing release after Stream Update!");
+		}
+	}
+
+	private void updateStreamDefinitionFromReleaseManifest(String streamName, String releaseManifest) {
+
+		List<SpringCloudDeployerApplicationManifest> appManifests =
+				new SpringCloudDeployerApplicationManifestReader().read(releaseManifest);
+
+		Map<String, SpringCloudDeployerApplicationManifest> appManifestMap = new HashMap<>();
+
+		for (SpringCloudDeployerApplicationManifest am : appManifests) {
+			String name = am.getSpec().getApplicationProperties().get("spring.cloud.dataflow.stream.app.label");
```

Comment:
Can refer constants instead of string here.

Reply:
We can do the same for other places where the string values are used to refer the Data Flow property keys.

Reply:
addressed